src/App.tsx

```diff
@@ -1,24 +1,176 @@
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const visibleTodos = useMemo(() => {
+    return (todos.filter((todo) => {
+      switch (filterBy) {
+        case SortType.ALL:
+          return todos;
+
+        case SortType.ACTIVE:
+          return !todo.completed;
+
+        case SortType.COMPLETED:
+          return todo.completed;
+
+        default:
+          return [];
+      }
+    })
+    );
+  }, [filterBy, todos]);
+
+  const noCompleteTodos = todos.some((todo) => todo.completed);
+
+  const loadTodosData = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setInputDisable(false);
+      setIsError(false);
+      setTempTodo(null);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setIsError(true);
+      setErrorMessage("Can't load data...");
+    }
+  };
+
+  useEffect(() => {
+    loadTodosData();
+  }, []);
+
+  const createNewTodo = (query: string): Todo => {
+
+    const newId = Math.max(...todos.map((todo) => todo.id + 1));
+
+    const newTodo = {
+      id: newId,
+      userId: USER_ID,
+      title: query,
+      completed: true,
+    };
+
+    return newTodo;
+  };
+
+  const dataTodo = {
+    userId: USER_ID,
+    title,
+    completed: false,
+  };
+
+  const sendTodo = async () => {
+    setTempTodo({ id: 0, ...dataTodo });
+
+    try {
+      await addTodo(createNewTodo(title));
+      loadTodosData();
+
+      setIsError(false);
+    } catch {
+      setErrorMessage('Unable to add a todo');
+      setTempTodo(null);
+      setIsError(true);
+    }
+  };
+
+  const handleFromSubmit = (event: React.FormEvent) => {
+    setInputDisable(true);
+
+    event.preventDefault();
+
+    sendTodo();
+    setTitle('');
+  };
+
+  const removeTodo = async (todoId: number) => {
+    setTempTodo({ id: 0, ...dataTodo });
+
+    try {
+      await onDelete(todoId);
+      loadTodosData();
+    } catch (error) {
+      setTempTodo(null);
+      setIsError(true);
+      setErrorMessage('Unable to delete a todo');
+    }
+  };
+
+  const clearCompleted = () => {
+    const clearCompletedTodos = todos.filter((todo) => todo.completed);
+
+    clearCompletedTodos.forEach((todo) => removeTodo(todo.id));
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          title={title}
+          setTitle={setTitle}
+          onSubmit={handleFromSubmit}
+          disable={inputDisable}
+        />
+
+        {todos.length > 0 && (
```

Comment:
Replace with `todos.length`, it will be the same as `todos.length > 0`

Reply:
not fixed
if any unnecessary things (number of todos length) renders with such code you can write it in this way too: `!!todos.length`